azure/datalake/store/lib.py

```diff
@@ -183,15 +192,19 @@ def refresh_token(self, authority=None):
 
         context = adal.AuthenticationContext(authority +
                                              self.token['tenant'])
-        if self.token.get('secret') and self.token.get('client'):
-            out = context.acquire_token_with_client_credentials(self.token['resource'], self.token['client'],
-                                                                self.token['secret'])
-            out.update({'secret': self.token['secret']})
-        else:
-            out = context.acquire_token_with_refresh_token(self.token['refresh'],
-                                                           client_id=self.token['client'],
-                                                           resource=self.token['resource'])
-            out.update({'refresh': out['refreshToken']})
+
+        @retry_decorator(retry_policy=retry_policy)
+        def f():
+                if self.token.get('secret') and self.token.get('client'):
+                    out = context.acquire_token_with_client_credentials(self.token['resource'], self.token['client'],
```

Comment:
minor:make it into 3 lines as you did below?

Reply:
Fixed.